go/ai/prompt_test.go

```diff
@@ -1148,3 +1148,65 @@ Hello!
 		t.Fatalf("Failed to execute prompt: %v", err)
 	}
 }
+
+func TestMultiMessagesRenderPrompt(t *testing.T) {
+	tempDir := t.TempDir()
+
+	mockPromptFile := filepath.Join(tempDir, "example.prompt")
+	mockPromptContent := `---
+model: test/chat
+description: A test prompt
+---
+<<<dotprompt:role:system>>>
+
+You are a pirate!
+
+<<<dotprompt:role:user>>>
+Hello!
```

Comment:
The test case is failing due to the string mismatch.

`dotprompt.ToMessages` trims all Unicode space chars except new lines, this means that the system prompt will look like this: `\nYou are a pirate\n` and the user message like this: `Hello!\n`

I think that, for the sake of the test case, it would be fine to trim these new lines.